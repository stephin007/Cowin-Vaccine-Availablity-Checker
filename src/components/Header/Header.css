@import url("https://fonts.googleapis.com/css2?family=Fira+Mono:wght@500&display=swap");

.MuiAppBar-colorPrimary {
  background-color: #caf7e3 !important;
}

.MuiIconButton-label {
  color: black !important;
}

.MuiButtonBase-root {
  font-family: "Fira Mono", monospace !important;
  font-weight: 600 !important;
}

.MuiButtonBase-root {
  color: black !important;
}
