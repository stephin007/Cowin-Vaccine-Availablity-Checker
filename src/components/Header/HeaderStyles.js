import styled from 'styled-components';

export const Button = styled.button`
    background: transparent;
    border-radius: 3px;
    border: 2px solid ${({theme}) => theme.btnColor};
    color: ${({theme}) => theme.btnColor};
    margin: 0 1em;
    padding: 0.25em 1em;
`