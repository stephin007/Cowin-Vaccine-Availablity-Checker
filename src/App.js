import React from "react";
import "./App.css";
import { BrowserRouter as Router, Switch, Route } from "react-router-dom";

import { ThemeProvider } from "styled-components";
import { useDarkMode } from "./components/useDarkMode";
import { lightTheme, darkTheme } from "./components/themes";
import { GlobalStyles } from "./components/globalStyles";

import { MuiThemeProvider, createMuiTheme } from "@material-ui/core/styles";

import Home from "./components/Home/Home";
import Header from "./components/Header/Header";
import About from "./components/About/About";
import Intro from "./components/Intro/Intro";

import CovidInfoMain from "./components/CovidInfo/CovidInfoMain";

const App = () => {
  const [theme, toggleTheme] = useDarkMode();
  const themeMode = theme === "light" ? lightTheme : darkTheme;
  const appliedTheme = createMuiTheme({
    palette: {
      type: theme === "light" ? "light" : "dark",
    },
  });

  return (
    <ThemeProvider theme={themeMode}>
      <MuiThemeProvider theme={appliedTheme}>
        <>
          <GlobalStyles />
          <Router>
            <div className="app">
              <Header theme={theme} toggleTheme={toggleTheme} />
              <Switch>
                <Route exact={true} path="/">
                  <Intro />
                </Route>
                <Route path="/vaccines" exact={true}>
                  <Home />
                </Route>
                <Route path="/covidinfo" exact={true}>
                  <CovidInfoMain />
                </Route>
                <Route path="/about" exact={true}>
                  <About />
                </Route>
              </Switch>
            </div>
          </Router>
        </>
      </MuiThemeProvider>
    </ThemeProvider>
  );
};

export default App;
