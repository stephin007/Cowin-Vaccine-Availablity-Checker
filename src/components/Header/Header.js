import React, { useState } from 'react';
import { makeStyles, useTheme } from '@material-ui/core/styles';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import IconButton from '@material-ui/core/IconButton';
import MenuIcon from '@material-ui/icons/Menu';
import MenuItem from '@material-ui/core/MenuItem';
import Menu from '@material-ui/core/Menu';
import useMediaQuery from '@material-ui/core/useMediaQuery';
import './Header.css';
import { Link } from 'react-router-dom';
import Brightness7Icon from '@material-ui/icons/Brightness7';
import Brightness2Icon from '@material-ui/icons/Brightness2';

const useStyles = makeStyles((theme) => ({
	root: {
		flexGrow: 1,
	},
	menuButton: {
		marginRight: theme.spacing(2),
	},
	title: {
		[theme.breakpoints.down('xs')]: {
			flexGrow: 1,
		},
	},
	headerOptions: {
		display: 'flex',
		flex: 1,
		justifyContent: 'flex-end',
	},
}));

export const Header = (props) => {
	const classes = useStyles();
	const [anchorEl, setAnchorEl] = useState(null);
	const open = Boolean(anchorEl);
	const theme = useTheme();
	const isMobile = useMediaQuery(theme.breakpoints.down('xs'));

	const handleMenu = (event) => {
		setAnchorEl(event.currentTarget);
	};

	return (
		<div className={classes.root}>
			<AppBar position="static" elevation={0} data-testId="header-appbar">
				<Toolbar className={props.theme === 'light' ? 'header' : 'headerDark'} data-testId="header-toolbar">
					<Typography
						variant="h6"
						style={{ marginTop: 10 }}
						className={classes.title}
						data-testId="header-typography"
					>
						<Link to="/">
							<img
								src="https://user-images.githubusercontent.com/71087810/117496553-d5af7b00-af94-11eb-84bb-913a1f386811.png"
								alt=""
								style={{ width: 100 }}
								className="logo__img"
							/>
						</Link>
					</Typography>

					{isMobile ? (
						<>
							<IconButton
								edge="start"
								className={classes.menuButton}
								color="inherit"
								aria-label="menu"
								onClick={handleMenu}
							>
								<MenuIcon className={props.theme === 'light' ? 'header__ham' : 'header__hamDark'} />
							</IconButton>
							<Menu
								id="menu-appbar"
								anchorEl={anchorEl}
								anchorOrigin={{
									vertical: 'top',
									horizontal: 'right',
								}}
								keepMounted
								transformOrigin={{
									vertical: 'top',
									horizontal: 'right',
								}}
								open={open}
								onClose={() => setAnchorEl(null)}
							>
								{props.theme === 'light' ? (
									<>
										<Link to="/about" style={{ padding: 0, textDecoration: 'none' }}>
											<MenuItem style={{ color: 'black' }}>About</MenuItem>
										</Link>
										<Link to="/covidinfo" style={{ padding: 0, textDecoration: 'none' }}>
											<MenuItem style={{ color: 'black' }}>COVID19 Information</MenuItem>
										</Link>
									</>
								) : (
									<>
										<Link
											to="/about"
											style={{
												padding: 0,
												textDecoration: 'none',
												color: 'white',
											}}
										>
											<MenuItem>About</MenuItem>
										</Link>
										<Link
											to="/covidinfo"
											style={{
												padding: 0,
												textDecoration: 'none',
												color: 'white',
											}}
										>
											<MenuItem>COVID19 Information</MenuItem>
										</Link>
									</>
								)}
								<a
									href="https://github.com/stephin007/Cowin-Vaccine-Availablity-Checker/"
									target="_blank"
									rel="noreferrer"
									style={{ textDecorationLine: 'none' }}
								>
									{props.theme === 'light' ? (
										<MenuItem style={{ color: 'black' }}>Contribute</MenuItem>
									) : (
										<MenuItem style={{ color: 'white' }}>Contribute</MenuItem>
									)}
								</a>{' '}
								<MenuItem onClick={props.toggleTheme}>
									{props.theme === 'light' ? (
										<Brightness7Icon style={{ marginLeft: 30, color: 'black' }} />
									) : (
										<Brightness2Icon style={{ marginLeft: 30, color: 'black' }} />
									)}
								</MenuItem>
							</Menu>
						</>
					) : (
						<div className={classes.headerOptions}>
							{props.theme === 'light' ? (
								<>
									<Link
										to="/about"
										style={{
											padding: 0,
											textDecoration: 'none',
											color: 'black',
										}}
									>
										<MenuItem>About</MenuItem>
									</Link>
									<Link
										to="/covidinfo"
										style={{
											padding: 0,
											textDecoration: 'none',
											color: 'black',
										}}
									>
										<MenuItem>COVID19 Information</MenuItem>
									</Link>
								</>
							) : (
								<>
									<Link
										to="/about"
										style={{
											padding: 0,
											textDecoration: 'none',
											color: 'white',
										}}
									>
										<MenuItem>About</MenuItem>
									</Link>
									<Link
										to="/covidinfo"
										style={{
											padding: 0,
											textDecoration: 'none',
											color: 'white',
										}}
									>
										<MenuItem>COVID19 Information</MenuItem>
									</Link>
								</>
							)}
							<a href="https://github.com/stephin007/Cowin-Vaccine-Availablity-Checker/">
								{props.theme === 'light' ? (
									<MenuItem style={{ color: 'black' }}>Contribute</MenuItem>
								) : (
									<MenuItem style={{ color: 'white' }}>Contribute</MenuItem>
								)}
							</a>
							<MenuItem onClick={props.toggleTheme} data-testId="header-toggleTheme">
								{' '}
								{props.theme === 'light' ? (
									<Brightness2Icon style={{ color: 'black' }} />
								) : (
									<Brightness7Icon style={{ color: 'white' }} />
								)}
							</MenuItem>
						</div>
					)}
				</Toolbar>
			</AppBar>
		</div>
	);
};

export default Header;
