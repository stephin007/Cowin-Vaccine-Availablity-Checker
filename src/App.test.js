import React from 'react';
import App from './App';
import { shallow } from 'enzyme';
import { Header } from './components/Header/Header';

describe('Rendering Header Component', () => {
	let header = shallow(<Header />);

	it('renders AppBar Component', () => {
		expect(header.find('[data-testId="header-appbar"]').exists()).toEqual(true);
	});

	it('renders toolbar component', () => {
		expect(header.find('[data-testId="header-toolbar"]').exists()).toEqual(true);
	});
	it('renders typography component', () => {
		expect(header.find('[data-testId="header-typography"]').exists()).toEqual(true);
	});
	it('renders toggle button component', () => {
		expect(header.find('[data-testId="header-toggleTheme"]').exists()).toEqual(true);
	});
});

