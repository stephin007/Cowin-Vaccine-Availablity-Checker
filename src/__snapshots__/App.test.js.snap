// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering Header Component renders correctly 1`] = `ShallowWrapper {}`;
